python/cudf_polars/cudf_polars/utils/config.py

```diff
@@ -5,209 +5,122 @@
-from typing import TYPE_CHECKING, Any
+import dataclasses
+from typing import TYPE_CHECKING, Literal
 
 if TYPE_CHECKING:
-    from collections.abc import MutableMapping, Sequence
-
     from typing_extensions import Self
 
+    import polars as pl
+
+
 __all__ = ["ConfigOptions"]
 
 
-class ConfigOptions:
-    """
-    GPUEngine configuration-option manager.
+@dataclasses.dataclass(frozen=True)
+class ParquetOptions:
+    """Configuration for the cudf-polars Parquet engine."""
 
-    This is a convenience class to help manage the nested
-    dictionary of user-accessible `GPUEngine` options.
-    """
+    chunked: bool = True
+    chunk_read_limit: int = 0
+    pass_read_limit: int = 0
 
-    __slots__ = ("_hash_value", "config_options")
-    _hash_value: int
-    config_options: dict[str, Any]
-    """The underlying (nested) config-option dictionary."""
 
-    def __init__(self, options: dict[str, Any]):
-        self.config_options = self.validate(copy.deepcopy(options))
+FALLBACK_MODE = Literal["warn", "raise"]
 
-    def set(self, name: str, value: Any) -> Self:
-        """
-        Set a user config option.
 
-        Nested dictionary keys should be separated by periods.
-        For example::
+STREAMING_EXECUTOR_PARQUET_DEFAULTS = {
+    "chunked": False,
+}
 
-            >>> options = options.set("parquet_options.chunked", False)
 
-        Parameters
-        ----------
-        name
-            Period-separated config name.
-        value
-            New config value.
-        """
-        options = config_options = copy.deepcopy(self.config_options)
-        keys = name.split(".")
-        for k in keys[:-1]:
-            assert isinstance(options, dict)
-            if k not in options:
-                options[k] = {}
-            options = options[k]
-        options[keys[-1]] = value
-        return type(self)(config_options)
-
-    def get(self, name: str, *, default: Any = None) -> Any:
-        """
-        Get a user config option.
+@dataclasses.dataclass(frozen=True, eq=True)
+class StreamingExecutor:
+    """
+    Configuration for the cudf-polars streaming executor.
 
-        Nested dictionary keys should be separated by periods.
-        For example::
+    Parameters
+    ----------
+    scheduler
+        The scheduler to use for the streaming executor.
 
-            >>> chunked = config_options.get("parquet_options.chunked")
+        "distributed" requires a Dask cluster to be running.
+    """
 
-        Parameters
-        ----------
-        name
-            Period-separated config name.
-        default
-            Default return value.
+    name: Literal["streaming"] = "streaming"
+    scheduler: Literal["synchronous", "distributed"] = "synchronous"
+    fallback_mode: FALLBACK_MODE = "warn"
+    max_rows_per_partition: int = 1_000_000
+    cardinality_factor: dict[str, float] = dataclasses.field(default_factory=dict)
+    parquet_blocksize: int = 1_000_000_000  # why isn't this a ParquetOption?
```

Comment:
I was a little confused why this shows up as an executor option, rather than a parquet option. I guess because the blocksize we want by default depends on the executor?

Reply:
It's actually because the `"in-memory"` executor doesn't support this option at all.

Reply:
Ah, makes sense then.